### ISSUE TITLE
Performance Degradation on MLPerf's GNMT implementation after converting to frozen graph 

### ISSUE THREAD

**Issue by brycearden** [tensorflow/tensorflow#33000]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): no
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): ('v1.14.0-rc1-22-gaf24dc91b5', '1.14.0')
- Python version: 2.7.15+
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source): `gcc (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0`
- CUDA/cuDNN version:
- GPU model and memory: NVIDIA Titan V

I am noticing a significant performance degradation in inference time when using the GNMT model provided for the MLPerf v0.5 benchmark with a trained model loaded from a checkpoint vs a trained model represented as a frozen graph. On the first iteration of inference (with batch_size = 6), the checkpoint graph takes ~2.9sec to perform inference while the frozen_graph implementation takes > 40sec. I would expect the frozen_graph to have better performance than the checkpoint model since my impression was that frozen graphs were meant for inference. 

Output from model loaded from checkpoint:
```
# Start decoding
  decoding to output /foobar/g_nmt-out
  infer_mode beam_search, beam_width 10, num translations per input 1.
  total iterations count 1.
infer time: 2.9462 secs
  done, num sentences 6, num translations per input 1, time 2s, Wed Oct  2 16:02:07 2019.
single_worker_inference time: 3.0597 secs
  bleu: 16.8
```

Output from model loaded from a frozen graph:
```
# Start decoding
  decoding to output /foobar/g_nmt-out
  infer_mode beam_search, beam_width 10, num translations per input 1.
  total iterations count 1.
infer time: 40.5799 secs
  done, num sentences 6, num translations per input 1, time 40s, Wed Oct  2 15:40:09 2019.
single_worker_inference time: 40.7543 secs
  bleu: 16.8
```

I've pushed up my own fork of the [mlperf/inference](https://github.com/mlperf/inference) repository with a branch that exposes this performance degradation [found here](https://github.com/brycearden/inference/tree/gnmt-frozen-perf-degradation). The fast / slow behavior can be toggled here, and can be reproduced by running `python run_task.py --run=accuracy`:

https://github.com/brycearden/inference/blob/gnmt-frozen-perf-degradation/v0.5/translation/gnmt/tensorflow/nmt/inference.py#L131-L160

Note that on the second iteration (after the caches warm up), the frozen graph still seems to be ~3x slower than the checkpoint model (7sec vs 2sec)

Any additional insight would be greatly appreciated.

**Comment by jvishnuvardhan** [comment]
@brycearden Can you please try the recently released `TF1.15.0rc2` and let us know how it progresses. Thanks!

**Comment by brycearden** [comment]
After trying on `1.15.0rc2` with Python 3.6.8, there seemed to be very little improvement:
```
❯ python3 
Python 3.6.8 (default, Aug 20 2019, 17:12:48)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
>>> print (tf.__version__)
1.15.0-rc2
```
Loading from checkpoint:
```
# Start decoding
  decoding to output /foo/g_nmt-out
  infer_mode beam_search, beam_width 10, num translations per input 1.
  total iterations count 1.
infer time: 2.9555 secs
  done, num sentences 6, num translations per input 1, time 2s, Thu Oct  3 20:07:45 2019.
single_worker_inference time: 3.0558 secs
  bleu: 16.8
```
Loading from the frozen graph:
```
# Start decoding
  decoding to output /goo/g_nmt-out
  infer_mode beam_search, beam_width 10, num translations per input 1.
  total iterations count 1.
infer time: 40.7798 secs
  done, num sentences 6, num translations per input 1, time 40s, Thu Oct  3 20:11:41 2019.
single_worker_inference time: 40.9498 secs
  bleu: 16.8
```

Reproducible using the link(s) provided above running `python3 run_task.py --run=accuracy`

**Comment by jvishnuvardhan** [comment]
@brycearden Sorry for the late response. Is this still an issue for you. Thanks!

**Comment by brycearden** [comment]
The GNMT tensorflow benchmark relies on `tf.contrib` heavily and would be a non-trivial amount of work to get this benchmark converted into TF2. However, this still reproduces on tensorflow `1.15.0`.

**Comment by tensorflowbutler** [comment]
It has been 25 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by brycearden** [comment]
yes

**Comment by jvishnuvardhan** [comment]
@brycearden Can you please try `TF1.15.3` and let us know if the issue persists or not. If the issue persists, can you please share a simple standalone code to reproduce the issue? Thanks!

Please close the issue if this was resolved for you. Thanks!
